File: crates/process_web/src/api/sharing_request_log.rs
use axum::extract::{Path, State};
use axum::routing::{get, post};
use axum::{Json, Router};
use serde::Deserialize;
use std::sync::Arc;
use ts_rs::TS;

use crate::api::common::*;
use crate::entity::sharing_request_log::Model;
use crate::service::sharing_request_log_service::SharingRequestLogService;
use crate::{bool_response, data_response, pagination_response};

#[derive(Deserialize, TS)]
#[ts(
    export,
    export_to = "ui/api/models/auto-generates/SharingRequestLogParams.ts",
    rename = "SharingRequestLogParams"
)]
pub struct ListParams {
    pub data_sharing_config_name: Option<String>,
}

pub fn set_routes() -> Router<Arc<AppState>> {
    let routes = Router::new()
        .route("/find_by_id/:id", get(find_by_id))
        .route("/list", post(list))
        .route("/add", post(add))
        .route("/update_by_id/:id", post(update_by_id))
        .route("/del/:id", get(del));

    routes
}

async fn find_by_id(
    state: State<Arc<AppState>>,
    Path(id): Path<i32>,
) -> anyhow::Result<ResJson<Model>, AppError> {
    let res = SharingRequestLogService::find_by_id(&state.conn, id).await;

    data_response!(res)
}

async fn list(
    state: State<Arc<AppState>>,
    Json(payload): Json<PaginationPayload<ListParams>>,
) -> anyhow::Result<ResJsonWithPagination<serde_json::Value>, AppError> {
    let res = SharingRequestLogService::list(
        &state.conn,
        payload.current,
        payload.page_size,
        payload.data,
    )
    .await;

    pagination_response!(res, payload.current, payload.page_size)
}
async fn add(
    state: State<Arc<AppState>>,
    Json(payload): Json<Model>,
) -> anyhow::Result<ResJson<Model>, AppError> {
    let res = SharingRequestLogService::add(&state.conn, payload).await;

    data_response!(res)
}

async fn update_by_id(
    state: State<Arc<AppState>>,
    Path(id): Path<i32>,
    Json(payload): Json<Model>,
) -> anyhow::Result<ResJson<Model>, AppError> {
    let res = SharingRequestLogService::update_by_id(&state.conn, id, payload).await;

    data_response!(res)
}

async fn del(
    state: State<Arc<AppState>>,
    Path(id): Path<i32>,
) -> anyhow::Result<ResJson<bool>, AppError> {
    let res = SharingRequestLogService::delete(&state.conn, id).await;
    bool_response!(res)
}

File: crates/process_web/ui/api/sharing_request_log.ts
import { SharingRequestLog } from "@/api/models/SharingRequestLog";
import {
  http_get,
  http_post,
  PaginationPayload,
  ResJson,
  ResJsonWithPagination,
} from "@/api/common";
import { SharingRequestLogParams } from "@/api/models/SharingRequestLogParams";

export const PREFIX = "/api/sharing_request_log";

export const FIND_BY_ID = `${PREFIX}/find_by_id/`;

export async function find_by_id(
  id: number
): Promise<ResJson<SharingRequestLog>> {
  return http_get(`${FIND_BY_ID}${id}`);
}

export const LIST = `${PREFIX}/list/`;
export async function list(
  payload: PaginationPayload<SharingRequestLogParams>
): Promise<ResJsonWithPagination<SharingRequestLog>> {
  return http_post(`${PREFIX}/list`, {
    body: JSON.stringify(payload),
  });
}

export const ADD = `${PREFIX}/add/`;
export async function add(
  payload: SharingRequestLog
): Promise<ResJson<SharingRequestLog>> {
  console.log(payload);
  return http_post(`${PREFIX}/add`, {
    body: JSON.stringify(payload),
  });
}

export const UPDATE_BY_ID = `${PREFIX}/update_by_id/`;
export async function update_by_id(
  id: string,
  payload: SharingRequestLog
): Promise<ResJson<SharingRequestLog>> {
  return http_post(`${PREFIX}/update_by_id/${id}`, {
    body: JSON.stringify(payload),
  });
}

export const DEL = `${PREFIX}/del/`;
export async function del(id: string): Promise<ResJson<boolean>> {
  return http_get(`${PREFIX}/del/${id}`);
}
